src/main/java/seedu/address/model/animal/AnimalContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.animal;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.model.feedtime.FeedTime;
+import seedu.address.model.medicalcondition.MedicalCondition;
+
+/**
+ * Tests that any of an {@code Animal}'s fields matches any of the keywords given.
+ */
+public class AnimalContainsKeywordsPredicate implements Predicate<Animal> {
+    private final List<String> keywords;
+
+    public AnimalContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    /**
+     * Checks if the keyword matches any of the strings in an animal's feeding times.
+     * @param feedTimes A set containing all of the animal's feeding times.
+     * @param word A given keyword.
+     * @return the boolean that checks if the keyword matches.
```

Comment:
nitpick: should capitalise the first letter of the return javadoc sentence. same for the method below